url8551e9ebc1_mishika05_Salesforce_DataGen_tgz-pJ8-org_json_TestJ8/src/Record.java

```diff
@@ -0,0 +1,337 @@
+		logger.info("Accessing records from result of query (getFromRecords())");
+		return ((String)((JSONArray) (orig.get("records"))).getJSONObject(0).get(label)).toString();
+	}
+	
+	/**
+     * posts record with structure of JSON object passed and into category
+     * of specified sObjectRec
+     * @param JSONObject representing structure of record, fields and values
+	 * @throws Exception 
+     */
+	protected boolean postRecord(JSONObject jobj) throws Exception {
+		//  logger.info("Attempting to post record (postRecord(JSONObject j))");
+		  HttpPost post = new HttpPost(my_sf.prop.environment + Constants.postUrl  + sObjectRec +"/");
+		 post.addHeader("Authorization", "Bearer " + my_sf.accessToken);
+		  post.setEntity(new StringEntity(jobj.toString(), 
+                  ContentType.create("application/json")));
+	        try {
+	            HttpResponse httpResponse = my_sf.httpclient.execute(post);
+	            String response = new String();
+	        	 HttpEntity responseEntity = httpResponse.getEntity();
+	        
+	        	 //error occurred
+	        	 if(responseEntity !=null) {
+	        	     response = EntityUtils.toString(responseEntity);
+	        	     if (response.contains("error") && !response.contains("success")) {
+	        	    	 logger.error(response);
+	        	    	 if (response.contains("INVALID_SESSION_ID")) {
+	 	        	    	my_sf.retrieveAccessToken();
+	 	        	     }
+	        	    	 return false;
+	        	     }
+	        	     else {
+	        	    	 JSONObject j = new JSONObject(response);
+	        	    	 String recId = (String) j.get("id");
+	        	    	 /*String name = "";
+	        	    	 JSONObject jj = queryRecords("Select Name from ADM_Work__c where Id = '" + j.get("id") + "'");
+	        	    	 String name = getFromRecords(jj, "Name");*/
+	        	    	 String recordName = "";
+	        	    	 if (uniqueNameField != "") {
+	        	    		 if (jobj.has(uniqueNameField)) {
+	        	    			 recIdsLogger.info(recId);
+	        	    			 recordName = jobj.getString(uniqueNameField);
+	        	    		 }
+	        	    	 }
+	        	    	 logger.info("Record " + recordName + " Posted - Post Successful");
+	        	     }	 
+	        	 }
+	        	 return true;
+	        } finally {
+	            post.releaseConnection();
+	        }		
+	}
+	
+	/**
+     * calls postRecord with structure of JSON object passed and into category
+     * of specified field sObjectRec, is post is unsuccessful, retries x num. of times
+     * @param JSONObject representing structure of record, fields and values
+	 * @throws Exception 
+     */
+	public boolean validatePost(JSONObject jobj) throws Exception {
+		//logger.info("Attempting to Validate Post");
+		int trial = 0;
+		boolean success = postRecord(jobj);
+		while (!success && trial < Constants.numTries) {
+			if (!success) {
+		    	 String recordName = "";
+    	    	 if (uniqueNameField != "")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.